src/Http/Routing/src/Internal/LinkGenerationDecisionTree.cs

```diff
@@ -91,7 +108,20 @@ public IList<OutboundMatchResult> GetMatches(RouteValueDictionary values, RouteV
                 {
                     if (criterion.Branches.TryGetValue(value ?? string.Empty, out var branch))
                     {
-                        Walk(results, values, ambientValues, branch, isFallbackPath);
+                        Walk(results, values, ambientValues, branch, quality + 1m);
```

Comment:
Change the quality modifier values to const fields

Reply:
obsolete, they are gone completely